tutorials/basic-topics/greedy.md

```diff
@@ -10,8 +10,120 @@ keywords:
   - algorithm
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@abhishek-sultaniya"/>
 
 ## Overview
 
-...
\ No newline at end of file
+Greedy is a strategy to solve various problems optimally.A greedy algorithm constructs a solution to the problem by always making a
+choice that looks the best at the moment. A greedy algorithm never takes back
+its choices, but directly constructs the final solution. For this reason, greedy
+algorithms are usually very efficient.
+
+Rules:
+
+1)Always pick the best choice at any step
+
+2)Works on the assumption that by picking a local optimum solution at each step we will reach a global optimum
+
+3)The greedy algorithm never reconsiders its choices
+
+```
+Example 1: You have coins of denomination Rs 10, Rs 5, and Rs 1.Find the minimum number of coins to get a total of 17 Rs.
+
+Solution: 4 coins: 10+5+1+1
+
+Example 2: You have coins of denomination Rs 10, Rs 7, and Rs 1.Find the minimum number of coins to get a total of 16 Rs.
+Solution: 4 coins: 7+7+1+1
+Solution if greedy was used : 7 coins: 10+1+1+1+1+1+1
```

Comment:
what problem is that?

Reply:
Ok,I will elaborate more on this.

This focuses on why we use greedy and how we approach it.
And second one shows directly and blindly we cant use greedy,first think and implement a efficent greedy before diectly applying it.

Reply:
I'm asking which LC problem is that. If it's not, remove it.

Reply:
Basically its a general thinking ideology,not any problem.

Reply:
https://leetcode.com/problems/coin-change/

This is LC problem

It is little difficult so I am not explaining it in beginning.